pkg/common/vclib/storagepod.go

```diff
@@ -134,13 +133,96 @@ func (spi *StoragePodInfo) ListFirstClassDisksInfo(ctx context.Context) ([]*Firs
 	return objs, nil
 }
 
+// GetFirstClassDiskInfoByName gets a specific first class disks (FCD) on this datastore backed by this StoragePodInfo
+func (spi *StoragePodInfo) GetFirstClassDiskInfoByName(ctx context.Context, diskName string) (*FirstClassDiskInfo, error) {
+	err := spi.PopulateChildDatastoreInfos(ctx, false)
+	if err != nil {
+		glog.Errorf("PopulateChildDatastoreInfos failed. Err: %v", err)
+		return nil, err
+	}
+
+	m := vslm.NewObjectManager(spi.Datacenter.Client())
+
+	for _, child := range spi.DatastoreInfos {
+		oids, err := m.List(ctx, child)
+		if err != nil {
+			glog.Errorf("Failed to list disks. Err: %v", err)
+			return nil, err
+		}
+
+		var ids []string
```

Comment:
Could drop this var and loop, just use `oids` as-is in the main loop.

Reply:
See previous comment